src/plugins/intel_cpu/CMakeLists.txt

```diff
@@ -306,14 +288,14 @@ cross_compiled_file(${TARGET_NAME}
         NAMESPACE   ov::Extensions::Cpu::XARCH
 )
 cross_compiled_file(${TARGET_NAME}
-        ARCH ${SOFTMAX_ARCH_LIST}
+        ARCH AVX512F AVX2 NEON_FP16 SVE ANY
```

Comment:
With that change in case `ENABLE_NEON_FP16=OFF` dispatch function will still look like:
```
    if (with_cpu_neon_fp16()) {
        return NEON_FP16::attn_softmax(a, a_dst, scale, alibi, attn_mask, causal_mask, select_nfltmax_at_0, len, total_size, precision, attn_mask_prec, dst_precision);
    }
    if (with_cpu_sve()) {
        return SVE::attn_softmax(a, a_dst, scale, alibi, attn_mask, causal_mask, select_nfltmax_at_0, len, total_size, precision, attn_mask_prec, dst_precision);
    }
    if (true) {
        return ANY::attn_softmax(a, a_dst, scale, alibi, attn_mask, causal_mask, select_nfltmax_at_0, len, total_size, precision, attn_mask_prec, dst_precision);
    }
```
The only difference is that NEON_FP16::attn_softmax is compiled w/o -march=armv8.2-a+fp16.

Reply:
Fixed

Please have a look.